### ISSUE TITLE
How to get the default 'dataIdFromObject' value?

### ISSUE THREAD

**Issue by MichaelDeBoey** [apollographql/apollo-client#1657]
I don't have a custom `dataIdFromObject` function in the constructor of the client, so all my values are stored as `__typename:id` by default.

Now I want to access the default `dataIdFromObject`, so I can use it for a `readFragment`, where I only have the id of the object.
I think there should be a better way then this, or is this the recommended way?
```js
const folder = apollo.readFragment({
  id: `Folder:${selectedFolderId}`,
  fragment: folderFragment,
});
```

If I ever want to have a custom `dataIdFromObject` function (or if you guys change the default implementation), I would have to change every `readFragment` in my codebase, which would not be something I want.

**Comment by helfer** [comment]
@MichaelDeBoey I think currently this function is exposed as `client.dataId`, it's just not documented. A PR to the docs would be great.

I think we should also expose it as `dataIdFromObject` in addition to `dataId`. If you'd like to make a PR for that, that would be great as well! 🙂

**Comment by MichaelDeBoey** [comment]
@helfer I'd love to help, but I'm completely new here, so if you could point me in the right direction on how to start, that would be great! 🙂

**Comment by helfer** [comment]
Sure! For the code PR, the only file you'd have to modify is `ApolloClient.ts`!

**Comment by MichaelDeBoey** [comment]
@helfer I don't know if it was that easy to add it, or did I miss some things? 😕

**Comment by helfer** [comment]
No, that's pretty much it. If you want, you can add two tests that make sure it's the function you'd expect (once default, once custom).

**Comment by MichaelDeBoey** [comment]
Yeah I was thinking about that, but I couldn't find the test for `dataId` so... 😕

**Comment by helfer** [comment]
It's really not worth testing imo.

**Comment by MichaelDeBoey** [comment]
If you want I can write a test for it, but if you don't think it's necessary, we can just leave it like this 🙂

**Comment by MichaelDeBoey** [comment]
The only thing we (i.e.: I) have to do is add the docs now, but can't seem to find them 😕

**Comment by helfer** [comment]
Maybe here? http://dev.apollodata.com/react/cache-updates.html#normalization

**Comment by MichaelDeBoey** [comment]
@helfer Shouldn't it be updated on all clients and not only `React`?

**Comment by helfer** [comment]
In theory yes, but we don't have those docs there now, and the vast majority of users are on React.

**Comment by MichaelDeBoey** [comment]
@helfer docs are added 🙂

**Comment by MichaelDeBoey** [comment]
@helfer Also made sure the docs would be available [here](http://dev.apollodata.com/react/api.html#ApolloClient) 🙂

**Comment by MichaelDeBoey** [comment]
Closed by #1663, apollographql/react-docs#221 & apollographql/core-docs#286
